src/test/java/com/bookappstore/service/impl/BookServiceImpTest.java

```diff
@@ -0,0 +1,282 @@
+        book.setAuthor(createBookRequestDto.getAuthor());
+        book.setDescription(createBookRequestDto.getDescription());
+        book.setPrice(createBookRequestDto.getPrice());
+        book.setIsbn(createBookRequestDto.getIsbn());
+        book.setCoverImage(createBookRequestDto.getCoverImage());
+        Long idTest = 1L;
+
+        Mockito.when(bookRepository.findById(idTest)).thenReturn(Optional.of(book));
+        Mockito.when(bookRepository.save(any(Book.class))).thenReturn(book);
+
+        BookDto expectedBookDto = createBookDto();
+        expectedBookDto.setTitle(createBookRequestDto.getTitle());
+        expectedBookDto.setAuthor(createBookRequestDto.getAuthor());
+        expectedBookDto.setDescription(createBookRequestDto.getDescription());
+        expectedBookDto.setPrice(createBookRequestDto.getPrice());
+        expectedBookDto.setIsbn(createBookRequestDto.getIsbn());
+        expectedBookDto.setCoverImage(createBookRequestDto.getCoverImage());
+
+        Mockito.when(bookMapper.toDto(any(Book.class))).thenReturn(expectedBookDto);
+
+        BookDto result = bookService.update(idTest, createBookRequestDto);
+
+        assertNotNull(result);
+        assertEquals(expectedBookDto, result);
+
+        verify(bookRepository).findById(idTest);
+        verify(categoryRepository).findByIdIn(createBookRequestDto.getCategoriesIds());
+        verify(bookRepository).save(book);
+        verify(bookMapper).toDto(book);
+    }
+
+    @Test
+    @DisplayName("Search books by parameters")
+    public void testSearchBooksByParameters() {
+        BookSearchParametersDto searchParams = createTestSearchParameters();
+
+        Specification<Book> specification = Mockito.mock(Specification.class);
+
+        List<Book> books = new ArrayList<>();
+        books.add(createBookWithParams(1L, "Test Title 1",
+                "Test Author 1",
+                BigDecimal.valueOf(19.99)));
+        books.add(createBookWithParams(2L,
+                "Test Title 2",
+                "Test Author 2",
+                BigDecimal.valueOf(29.99)));
+
+        List<BookDto> bookDtos = List.of(
+                createBookDtoWithParams("Test Title 1", "Test Author 1", BigDecimal.valueOf(19.99)),
+                createBookDtoWithParams("Test Title 2", "Test Author 2", BigDecimal.valueOf(29.99))
+        );
+
+        Mockito.when(bookSpecificationBuilder.build(searchParams)).thenReturn(specification);
+        Mockito.when(bookRepository.findAll(specification)).thenReturn(books);
+        Mockito.when(bookMapper.toDtoList(books)).thenReturn(bookDtos);
+
+        List<BookDto> result = bookService.search(searchParams);
+
+        assertNotNull(result);
+        assertEquals(bookDtos.size(), result.size());
+
+        verify(bookSpecificationBuilder).build(searchParams);
+        verify(bookRepository).findAll(specification);
+        verify(bookMapper).toDtoList(books);
+    }
+
+    private Category createCategory(Long id, String name) {
```

Comment:
I really like that in this class you put the creation of test objects into private methods, you can apply this to other test classes.